### ISSUE TITLE
stage 2 build memory leak

### ISSUE THREAD

**Issue by AliChraghi** [ziglang/zig#9794]
i cannot build stage 2 because a lot of leak.
**OS: Ubuntu 21.04**
**RAM: 8GB DDR3**
**Zig Version: 0.9.0-dev.1076+d2b5105f5**
i have runned `zig build` in the latest version clone. but my laptop hangs and cannot continue building. (screenshots are before full usage of memory because if don't kill the build process i should restart laptop
![Screenshot-20210918210932-1463x529](https://user-images.githubusercontent.com/63465728/133896109-34d00c21-495d-4b70-afc3-8a7386dbb6f7.png)
and by @kubkon guide runned with `zig build -Dskip-install-lib-files -Dskip-non-native` still leaks
![Screenshot-20210918211405-1483x526](https://user-images.githubusercontent.com/63465728/133896128-a03ad47a-483a-48f3-860f-d44c1c1bab34.png)

**Comment by g-w1** [comment]
Duplicate of https://github.com/ziglang/zig/issues/6485

**Comment by kubkon** [comment]
@ifreund @g-w1 is this really a duplicate? Note that the OP is trying to build self-hosted using a prebuilt binary and not build stage1 (or do we depend on zig0.o for self-hosted too?)

**Comment by ifreund** [comment]
It's the same root cause, `zig0.o` is a subset of the code of self-hosted.

**Comment by kubkon** [comment]
Gotcha, thanks for clarifying Isaac!
